cli/flox-activations/src/gen_rc/bash.rs

```diff
@@ -0,0 +1,203 @@
+        stmts.push(set_exported_unexpanded(
+            "FLOX_ENV_CACHE",
+            flox_env_cache.display().to_string(),
+        ));
+    } else {
+        stmts.push(unset("FLOX_ENV_CACHE"));
+    }
+
+    if let Some(flox_env_project) = &args.flox_env_project {
+        stmts.push(set_exported_unexpanded(
+            "FLOX_ENV_PROJECT",
+            flox_env_project.display().to_string(),
+        ));
+    } else {
+        stmts.push(unset("FLOX_ENV_PROJECT"));
+    }
+
+    if let Some(description) = &args.flox_env_description {
+        stmts.push(set_exported_unexpanded("FLOX_ENV_DESCRIPTION", description));
+    } else {
+        stmts.push(unset("FLOX_ENV_DESCRIPTION"));
+    }
+
+    stmts.push(set_exported_unexpanded(
+        "_flox_activate_tracer",
+        &args.flox_activate_tracer,
+    ));
+
+    // Set the prompt if we're in an interactive shell.
+    let set_prompt_path = args.activate_d.join("set-prompt.bash");
+    stmts.push(
+        format!(
+            "if [ -t 1 ]; then source '{}'; fi;",
+            set_prompt_path.display()
+        )
+        .to_stmt(),
+    );
+
+    // We already customized the PATH and MANPATH, but the user and system
+    // dotfiles may have changed them, so finish by doing this again.
+    // Use generation time _FLOX_ENV because we want to guarantee we activate the
+    // environment we think we're activating. Use runtime FLOX_ENV_DIRS to allow
+    // RC files to perform activations.
+    stmts.push(format!(
+        r#"eval "$('{}' set-env-dirs --shell bash --flox-env "{}" --env-dirs "${{FLOX_ENV_DIRS:-}}")";"#,
+        args.flox_activations.display(),
+        args.flox_env.display()
+    ).to_stmt());
+
+    stmts.push(format!(
+        r#"eval "$('{}' fix-paths --shell bash --env-dirs "$FLOX_ENV_DIRS" --path "$PATH" --manpath "${{MANPATH:-}}")";"#,
+        args.flox_activations.display()
+    ).to_stmt());
+
+    stmts.push(format!(
+        r#"eval "$('{}' profile-scripts --shell bash --already-sourced-env-dirs "${{_FLOX_SOURCED_PROFILE_SCRIPTS:-}}" --env-dirs "${{FLOX_ENV_DIRS:-}}")";"#,
+        args.flox_activations.display()
+    ).to_stmt());
+
+    // Disable command hashing to allow for newly installed flox packages
+    // to be found immediately. We do this as the very last thing because
+    // python venv activations can otherwise return nonzero return codes
+    // when attempting to invoke `hash -r`.
+    stmts.push("set +h".to_stmt());
+
+    // Disable trace mode if it was enabled
+    if args.flox_activate_tracelevel >= 2 {
+        stmts.push("set +x".to_stmt());
+    }
+
+    if let Some(path) = args.clean_up.as_ref() {
+        stmts.push(format!("rm '{}';", path.display()).to_stmt());
+    }
+
+    // N.B. the output of these scripts may be eval'd with backticks which have
+    // the effect of removing newlines from the output, so we must ensure that
+    // the output is a valid shell script fragment when represented on a single line.
```

Comment:
suggestion nonblocking: seems like this comment got relocated in the wrong spot and would be more helpful on the generate functions

Reply:
Fixed